### PR TITLE
Add modern components from the Sphinx ecosystem

### PR DESCRIPTION
**Pull request by amotl**
### About

This patch adds a few modern components from the Sphinx ecosystem to the theme, and demonstrates how they are used. Thanks to all the authors for your excellent work.

- [myst-parser](https://pypi.org/project/myst-parser), see also [Introduction: MyST Markdown - Markedly Structured Text :writing_hand:](https://community.panodata.org/t/technical-advancements-in-sphinx/278#myst-markdown-markedly-structured-text-writing_hand-8).
- [sphinx-design](https://pypi.org/project/sphinx-design/)
- [sphinx-design-elements](https://pypi.org/project/sphinx-design-elements/)
- [sphinx-inline-tabs](https://pypi.org/project/sphinx-inline-tabs/)
- [sphinx-subfigure](https://pypi.org/project/sphinx-subfigure)
- [sphinx-togglebutton](https://pypi.org/project/sphinx-togglebutton)
- [sphinxcontrib-mermaid](https://pypi.org/project/sphinxcontrib-mermaid)

### Preview
The preview of the demonstration pages can be inspected at RTD: [CrateDB docs theme » Preview](https://crate-docs-theme--378.org.readthedocs.build/en/378/). There is a dedicated section demonstrating how to use the corresponding modern Sphinx components, elements, and styles.

https://crate-docs-theme--378.org.readthedocs.build/en/378/#modernized

### CODE REVIEW

**Comment by matkuliak** [comment]
Hello @amotl  this is great, thank you!
I noticed you suggest removal of `sphinx_copybutton`. Why is that? Is it part of some of the newly added components?

**Comment by amotl** [comment]
> I noticed you suggest removal of `sphinx_copybutton`.

Please add an inline comment on the PR where you discovered this spot.

File: src/crate/theme/rtd/conf/__init__.py [hunk 1]
```diff
@@ -30,13 +30,25 @@
...
+    "sphinx_subfigure",
+    "sphinx_togglebutton",
     "sphinx.ext.intersphinx",
+    "sphinx.ext.todo",
+    "sphinx.ext.ifconfig",
+    "sphinx.ext.graphviz",
+    "sphinxcontrib.mermaid",
     "sphinxcontrib.plantuml",
     "sphinxext.opengraph",
-    "sphinx_copybutton",
```
Comment (matkuliak) [inline src/crate/theme/rtd/conf/__init__.py:37]: > I noticed you suggest removal of sphinx_copybutton.
>> That would not have been intended. Please add an inline comment on the PR where you discovered this spot.

Right here, @amotl

Comment (amotl) [reply src/crate/theme/rtd/conf/__init__.py:37]: Yeah. List is sorted alphabetically now. ;]

**Review by seut — APPROVED** [review]
Great! 👍

**Review by hlcianfagna — APPROVED** [review]
FYI @hammerhead , @karynzv

File: docs/myst-overview.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+**Example**
+
+> ![image|297x168](https://global.discourse-cdn.com/business7/uploads/crate/original/1X/d61bae27e7ce2ca04ab7ba78954153b885f2548e.png)
+
+
+## Overview
+
+### Inline quotes / citations
+
+> This text should be quoted.
```
Comment (amotl) [inline docs/myst-overview.md:34]: It looks like the markup instruction to quote/cite text does not render properly yet. /cc @msbt 

image

-- https://crate-docs-theme--378.org.readthedocs.build/en/378/myst-overview.html#inline-quotes-citations

Comment (amotl) [reply docs/myst-overview.md:34]: Exactly. It is written like
```
> Franz jagt im komplett verwahrlosten Taxi quer durch Bayern.
```

and usually renders like
> Franz jagt im komplett verwahrlosten Taxi quer durch Bayern.

Comment (matkuliak) [reply docs/myst-overview.md:34]: I guess this is the issue:
- https://github.com/crate/crate-docs-theme/blob/main/src/crate/theme/rtd/crate/static/css/crateio-rtd.css#L27
- https://github.com/crate/crate-docs-theme/blob/main/src/crate/theme/rtd/crate/static/css/components.css#L187

Just a missing border, right?

Comment (amotl) [reply docs/myst-overview.md:34]: Looks good, I think you found the right spot, thanks. 🙇

Comment (msbt) [reply docs/myst-overview.md:34]: @matkuliak @amotl https://github.com/crate/crate-docs-theme/blob/main/src/crate/theme/rtd/crate/static/css/crateio-rtd.css#L27-L31 is indeed the main issue, if we remove those lines, the quote works. But if you have a paragraph wrapped in there, we might also need to override the `margin-bottom`, else it won't be centered vertically:

image

Should I whip something up? And can you check if we're actually using the current `blockquote` somewhere in the docs @matkuliak?

Comment (amotl) [reply docs/myst-overview.md:34]: Thank you for submitting GH-380, aiming to fix this.

Comment (amotl) [reply docs/myst-overview.md]: I think this works well now, right?

- https://crate-docs-theme--378.org.readthedocs.build/en/378/rst/misc.html
- https://crate-docs-theme--378.org.readthedocs.build/en/378/myst/misc.html

Thank you again.

File: docs/rst/misc.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+Inline quotes / citations
+=========================
+
+    This text should be quoted.
+
+Todo items
+==========
+.. todo::
+
+    Foo bar baz.
```
Comment (amotl) [inline docs/rst/misc.rst:14]: Apparently, the `.. todo` directive renders itself with a bit too much of a left padding or margin. Maybe we can do something about it on a future iteration.

image


-- https://crate-docs-theme--378.org.readthedocs.build/en/378/rst/misc.html
-- https://crate-docs-theme--378.org.readthedocs.build/en/378/myst/misc.html

Comment (amotl) [reply docs/rst/misc.rst:14]: Created GH-381 in order to track this item.

**Pull request merged**